contracts/mainnet/connectors/compound/v3/main.sol

```diff
@@ -0,0 +1,845 @@
+			"invalid market/token address"
+		);
+
+		bool isEth = token == ethAddr;
+		address token_ = isEth ? wethAddr : token;
+
+		TokenInterface tokenContract = TokenInterface(token_);
+
+		uint256 initialBal = getAccountSupplyBalanceOfAsset(
+			address(this),
+			market,
+			token_
+		);
+
+		amt_ = amt_ == uint256(-1) ? initialBal : amt_;
+
+		if (token_ == getBaseToken(market)) {
+			uint256 balance = CometInterface(market).balanceOf(address(this));
+			if (balance > 0) {
+				require(amt_ <= balance, "withdraw-amt-greater-than-supplies");
+			}
+		}
+
+		CometInterface(market).withdraw(token_, amt_);
+
+		uint256 finalBal = getAccountSupplyBalanceOfAsset(
+			address(this),
+			market,
+			token_
+		);
+
+		amt_ = sub(initialBal, finalBal);
+
+		convertWethToEth(isEth, tokenContract, amt_);
+
+		setUint(setId, amt_);
+
+		eventName_ = "LogWithdraw(address,address,uint256,uint256,uint256)";
+		eventParam_ = abi.encode(market, token, amt_, getId, setId);
+	}
+
+	/**
+	 * @dev Withdraw base/collateral asset and transfer to 'to'.
+	 * @notice Withdraw base token or deposited token from Compound on behalf of an address and transfer to 'to'.
+	 * @param market The address of the market.
+	 * @param token The address of the token to be withdrawn. (For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param to The address to which the borrowed assets are to be transferred.
+	 * @param amt The amount of the token to withdraw. (For max: `uint256(-1)`)
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of tokens withdrawn.
+	 */
+	function withdrawTo(
+		address market,
+		address token,
+		address to,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	)
+		public
+		payable
+		returns (string memory eventName_, bytes memory eventParam_)
+	{
+		(uint256 amt_, uint256 setId_) = _withdraw(
+			BorrowWithdrawParams({
+				market: market,
+				token: token,
+				from: address(0),
+				to: to,
+				amt: amt,
+				getId: getId,
+				setId: setId
+			})
+		);
+
+		eventName_ = "LogWithdrawTo(address,address,address,uint256,uint256,uint256)";
+		eventParam_ = abi.encode(market, token, to, amt_, getId, setId_);
+	}
+
+	/**
+	 * @dev Withdraw base/collateral asset from an account and transfer to 'to'.
+	 * @notice Withdraw base token or deposited token from Compound from an address and transfer to 'to'.
+	 * @param market The address of the market.
+	 * @param token The address of the token to be withdrawn. (For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param from The address from where asset is to be withdrawed.
+	 * @param to The address to which the borrowed assets are to be transferred.
+	 * @param amt The amount of the token to withdraw. (For max: `uint256(-1)`)
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of tokens withdrawn.
+	 */
+	function withdrawFromUsingManager(
+		address market,
+		address token,
+		address from,
+		address to,
```

Comment:
@Richa-iitr @pradyuman-verma I am thinking to remove this. Since we are gonna withdraw DSA most of the times. If we need to send it to another address, we can use withdraw spell from basic connector.

Or we can have two variations of this function, one with `to` and another without `to`

Reply:
So from will always be `dsa`?  Then `withdrawTo` can be used.

Reply:
@Richa-iitr `from` will be an external position. DSA won't be `from`.

Eg: if your EOA has collateral if you need to withdraw them and deposit into DSA.

Reply:
done